crates/sui-types/src/messages.rs

```diff
@@ -49,7 +49,8 @@ pub enum CallArg {
     Pure(Vec<u8>),
     // an object
     Object(ObjectArg),
-    // TODO support more than one object (object vector of some sort)
+    // a vector of objects
+    ObjVec(Vec<ObjectArg>),
```

Comment:
From an expressivity standpoint, I think what we really want here is `Vec<CallArg>`--folks have also asked to pass in vectors of addresses, integers, and object ID's. Is there a reason not to opt for this more generic approach?

I understand that this schema can also represent bad heterogenous vectors (e.g., a vector containing a mix of object args and ints), but I think that will be caught in a later phase.

Reply:
> folks have also asked to pass in vectors of addresses, integers, and object ID's. Is there a reason not to opt for this more generic approach?

I did not put that much thought into supporting other types of vectors (yet), but since vectors of integers are already supported actually (I added a test for that in this PR), I thought that perhaps other vectors could be supported the same way, that it is via the `Pure` variant. The schema using `Vec<Box<CallArg>>` (I don't think `Vec<CallArg>` would work) also implies nested vectors, which we probably won't support. In summary, I thought of this PR as exclusively attacking the vector-of-objects problem and doing a necessary refactoring when working on PRs for other types of vectors. as I wasn't 100% sure which way it would go.

Reply:
Makes sense--I didn't realize pure vectors were already supported! This approach makes to me given that.